pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -181,7 +181,7 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*v1.Pod) *Results {
 			delete(errors, k)
 		}
 	}
-	return &Results{
+	return Results{
```

Comment:
Do we have benchmarking on this change or a heap profile? With results no longer being a pointer, this means that every time that it's passed around the inner values of the struct are copied. That should be fine as long as the things that are being stored in results are maps, arrays, and pointers

Reply:
I'll definitely be doing benchmarking before we make this change. But yes, all the inner values are maps and arrays, so we should be fine here. It's going from 1 pointer to 3 pointers. 

```
// Results contains the results of the scheduling operation
type Results struct {
	NewNodeClaims []*NodeClaim
	ExistingNodes []*ExistingNode
	PodErrors     map[*v1.Pod]error
}
```

Reply:
Sounds good to me. Interestingly enough: Slices are struct values under the hood that point to a combination of a pointer and a length. Not that that matters in this context for memory utilization, just thought it was interesting that it's slightly different from what I expected: https://go.dev/blog/slices